test/CompanyService.Business.UnitTests/Commands/Department/CreateDepartmentCommandTests.cs

```diff
@@ -89,108 +66,188 @@ public void OneTimeSetUp()
+            _autoMock
+                .Setup<IAccessValidator, bool>(x => x.HasRights(Rights.AddEditRemoveDepartments))
                 .Returns(false);
 
             Assert.Throws<ForbiddenException>(() => _command.Execute(_request));
-            _companyRepositoryMock.Verify(x => x.Get(It.IsAny<GetCompanyFilter>()), Times.Never);
-            _repositoryMock.Verify(x => x.CreateDepartment(It.IsAny<DbDepartment>()), Times.Never);
+
+            _autoMock.Verify<ICompanyRepository, DbCompany>(
+                x => x.Get(It.IsAny<GetCompanyFilter>()),
+                Times.Never);
+
+            _autoMock.Verify<IDepartmentRepository, Guid>(
+                x => x.CreateDepartment(It.IsAny<DbDepartment>()),
+                Times.Never);
+        }
+
+        [Test]
+        public void ShouldThrowValidationException()
+        {
+            _autoMock
+                .Setup<ICreateDepartmentRequestValidator, bool>(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                .Returns(false);
+
+            Assert.Throws<ValidationException>(() => _command.Execute(_request));
+
+            _autoMock.Verify<ICompanyRepository, DbCompany>(
+                x => x.Get(It.IsAny<GetCompanyFilter>()),
+                Times.Never);
+
+            _autoMock.Verify<IDepartmentRepository, Guid>(
+                x => x.CreateDepartment(It.IsAny<DbDepartment>()),
+                Times.Never);
         }
 
-        /*[Test]
-        public void ShouldThrowArgumentNullExceptionWhenDepartmentRequestIsNull()
+        [Test]
+        public void ShouldThrowExcWhenCompanyDoesntExist()
         {
+            var expected = new OperationResultResponse<Guid>
+            {
+                Status = OperationResultStatusType.Failed,
+                Errors = { "Company does not exist, please create company." }
+            };
 
-            _companyRepositoryMock
-                .Setup(x => x.Get(null))
-                .Returns(new DbCompany { Id = _companyId });
+            _autoMock
+                .Setup<ICompanyRepository, DbCompany>(x => x.Get(null))
+                .Returns((DbCompany)null);
 
-            _mapperMock
-                .Setup(x => x.Map(null, _companyId))
-                .Throws(new ArgumentNullException());
+            Assert.AreEqual(expected.Errors, _command.Execute(_request).Errors);
+            _autoMock.Verify<ICompanyRepository, DbCompany>(
+                x => x.Get(It.IsAny<GetCompanyFilter>()),
+                Times.Once);
 
-            Assert.Throws<ArgumentNullException>(() => _command.Execute(null));
-            _companyRepositoryMock.Verify(x => x.Get(null), Times.Once);
-            _repositoryMock.Verify(x => x.CreateDepartment(_dbDepartment), Times.Never);
-        }*/
+            _autoMock.Verify<IDepartmentRepository, Guid>(
+                x => x.CreateDepartment(It.IsAny<DbDepartment>()),
+                Times.Never);
```

Comment:
you can add verify for DoesNameExist and mapper

Reply:
done